drivers/dma/dma_arc_hs.c

```diff
@@ -0,0 +1,1011 @@
+/*
+ * Copyright (c) 2025 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/barrier.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/atomic.h>
+
+#define DT_DRV_COMPAT snps_designware_dma_arc_hs
+
+#ifdef CONFIG_DMA_64BIT
+#define dma_addr_t uint64_t
+#else
+#define dma_addr_t uint32_t
+#endif
+
+/* ARC DMA Auxiliary Register Definitions */
+#define DMA_AUX_BASE     (0xd00)
+#define DMA_C_CTRL_AUX   (0xd00 + 0x0)
+#define DMA_C_CHAN_AUX   (0xd00 + 0x1)
+#define DMA_C_SRC_AUX    (0xd00 + 0x2)
+#define DMA_C_SRC_HI_AUX (0xd00 + 0x3)
+#define DMA_C_DST_AUX    (0xd00 + 0x4)
+#define DMA_C_DST_HI_AUX (0xd00 + 0x5)
+#define DMA_C_ATTR_AUX   (0xd00 + 0x6)
+#define DMA_C_LEN_AUX    (0xd00 + 0x7)
+#define DMA_C_HANDLE_AUX (0xd00 + 0x8)
+#define DMA_C_STAT_AUX   (0xd00 + 0xc)
+
+#define DMA_S_CTRL_AUX      (0xd00 + 0x10)
+#define DMA_S_BASEC_AUX(ch) (0xd00 + 0x83 + ((ch) * 8))
+#define DMA_S_LASTC_AUX(ch) (0xd00 + 0x84 + ((ch) * 8))
+#define DMA_S_STATC_AUX(ch) (0xd00 + 0x86 + ((ch) * 8))
+#define DMA_S_DONESTATD_AUX(d)                                                                     \
+	(0xd00 + 0x20 + (d)) /* Descriptor selection. Each D stores descriptors d*32 +: 32 */
+#define DMA_S_DONESTATD_CLR_AUX(d) (0xd00 + 0x40 + (d))
+
+/* ARC DMA Attribute Flags */
+#define ARC_DMA_NP_ATTR       (1 << 3) /* Enable non posted writes */
+#define ARC_DMA_SET_DONE_ATTR (1 << 0) /* Set done without triggering interrupt */
+
+/* ARC Auxiliary Register Access Functions */
+static inline void arc_write_aux(uint32_t reg, uint32_t val)
+{
+	__asm__ volatile("sr %0, [%1]" : : "r"(val), "r"(reg) : "memory");
+}
```

Comment:
It would be good to check if Zephyr has any existing functions for doing the same thing.

Same with `arc_read_aux()`